codex/conf.nim

```diff
@@ -444,47 +448,63 @@ proc updateLogLevel*(logLevel: string) {.upraises: [ValueError].} =
         warn "Unrecognized logging topic", topic = topicName
 
 proc setupLogging*(conf: CodexConf) =
+  proc errorFallback(msg: string) =
+    try:
+      stderr.write msg & "\n"
+    except IOError:
+      echo msg
+
   when defaultChroniclesStream.outputs.type.arity != 2:
     warn "Logging configuration options not enabled in the current build"
   else:
     proc noOutput(logLevel: LogLevel, msg: LogOutputStr) = discard
-    proc writeAndFlush(f: File, msg: LogOutputStr) =
-      try:
-        f.write(msg)
-        f.flushFile()
-      except IOError as err:
-        logLoggingFailure(cstring(msg), err)
 
-    proc stdoutFlush(logLevel: LogLevel, msg: LogOutputStr) =
-      writeAndFlush(stdout, msg)
+    # Managing the file ourselves may seem distasteful, but chronicles does not
```

Comment:
This doesn't sound correct as per chronicles own docs - https://github.com/status-im/nim-chronicles

@zah could you please clarify what would be the proper way of handling logging to either a file, stdout or both?

Reply:
Ahah your opinion is just as valuable, @AuHau, just didn't think of dragging too many people into such a minor issue. At this point I'll wait for @zah's take, but what I could grok from docs + code + behavior is that these things are statically defined and that you can't mute a file sink - if it's defined, it will want to output.

Reply:
Ahah your opinion is just as valuable, @AuHau, just didn't think of dragging too many people into such a minor issue. At this point I'll wait for @zah's take, but what I could grok from docs + code + behavior is that these things are statically defined and that you can't mute a file sink - if it's defined, it will want to output. The only thing you get to define dynamically is the name of the output file. You could in principle use that and send it to `/dev/null` to mute it, but that's not portable.

The other option would be having a dynamic sink which is also a file sink (we use a null writer trick to mute dynamic sinks), but I haven't found a way to do that - it's either one or the other. And while this could be built on top of a dynamic sink, the resulting code will be more complicated than what's currently there.